ccdproc/image_collection.py

```diff
@@ -694,10 +700,12 @@ def _generator(self, return_type,
 
             file_name = path.basename(full_path)
 
+            extension_index = hdulist.index_of(self._extension)
+
             return_options = {
-                    'header': lambda: hdulist[0].header,
-                    'hdu': lambda: hdulist[0],
-                    'data': lambda: hdulist[0].data,
+                    'header': lambda: hdulist[extension_index].header,
+                    'hdu': lambda: hdulist[extension_index],
+                    'data': lambda: hdulist[extension_index].data,
                     'ccd': lambda: fits_ccddata_reader(full_path, **ccd_kwargs)
```

Comment:
Even though it's more complicated here I think it shouldn't be silently ignored in the `'ccd'`-case. What makes it complicated is that it's also possible to explicitly define a `hdu` by passing it as item-pair in `ccd_kwargs`.